vortex-array/src/array/varbin/compute/take.rs

```diff
@@ -49,7 +49,7 @@ fn take<I: NativePType, O: NativePType>(
         return Ok(take_nullable(dtype, offsets, data, indices, v));
     }
 
-    let mut builder = VarBinBuilder::<I>::with_capacity(indices.len());
+    let mut builder = VarBinBuilder::<O>::with_capacity(indices.len());
```

Comment:
previously, take(VarBinArray) was yielding a new VarBinArray with offset that had ptype `indices.ptype`, we actually wanted them to maintain their original ptype.

This is also why the arrow representation of string arrays often switched from Utf8 -> LargeUtf8 after doing a take, as in the pytest above